lib/Element.js

```diff
@@ -245,13 +246,13 @@ Element.prototype.getBoundingClientRect = function () {
   return this.component.getBoundingClientRect()
 }
 
-Element.prototype.toReact = function (idPrefix, index) {
-  idPrefix = idPrefix || 'faux-dom'
-  index = index || 0
-
+Element.prototype.toReact = function (id) {
   var props = assign({}, this.props)
   props.style = assign({}, props.style)
-  props.key = isUndefined(props.key) ? idPrefix + '-' + index : props.key
+
+  if (isUndefined(props.key) && id) {
```

Comment:
I'm really not sure about _ prefixed methods + getters/setters. I think
there's a much simpler way. I'll reply with a PR or something soon.

On 13 September 2016 at 19:30, FunkMonkey <notifications@github.com> wrote:

> In lib/Element.js
> <https://github.com/Olical/react-faux-dom/pull/73#discussion_r78617435>:
>
> >    var props = assign({}, this.props)
> >    props.style = assign({}, props.style)
> > -  props.key = isUndefined(props.key) ? idPrefix + '-' + index : props.key
> > +
> > +  if (isUndefined(props.key) && id) {
>
> @philpl <https://github.com/philpl> I think we're also missing a check
> for this._key and need to set it to props.key if necessary. Otherwise
> this._key has never been used. Right?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Olical/react-faux-dom/pull/73/files/8a5f8837612a0ee97ba73a56073f2fe837c4a925..120f8f10bff6468fec5f92c41e3b7c765c2cd407#r78617435>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AATPXav-Kt_k3RuT4IXJwkgCoxssSJk3ks5qpuvggaJpZM4J33UN>
> .
>